cdap-ui/app/services/app-level-loading-service.js

```diff
@@ -38,4 +38,14 @@ angular.module(PKG.name + '.services')
       }
     };
 
+    this.hideLoadingIconImmediate = function(){
+      if (!deferred) {
+        return $q.when(true);
+      } else {
+        EventPipe.emit('hideLoadingIcon.immediate');
+        deferred.resolve(true);
+        deferred = null;
```

Comment:
We should probably see if we can avoid resetting `deferred` to `null`. We can check the `pending-state` of the promise and if its is resolved then we can create a new promise (and let the browser GC the already resolved promise).